### PR TITLE
Dining redesign: open until subtitle, updated day chips, availability card redesign, improved chart and menus

### PR DESCRIPTION
**Pull request by kaushikravikumar**
### Summary 

This pull request adds a toolbar that utilizes the api endpoint facilityHours in order to fetch the operating hours of a facility. Using this info, the user sees in the subtitle of the toolbar whether it's open or closed and exactly when it is open until or when it is closed until. **Right Now, entire subtitle is red or green depending on whether it is closed or open. Will work on implementing it such that it matches the design on Figma, where only the word open or closed is colored in.**

This pull request is the first step towards the dining redesign effort.

- [x] Changed Day Chips Design
- [x] Toolbar with Open/Close Info for facility
- [x] Change Availability Card
- [x] Clean up Menu design
- [x] Gray out closed hours of Historical data chart

### Test Plan 

- I created a list inside the facilityHoursOnResponse function definition inside of the checkFacilityIsOpen() function inside the FacilityInfoPage Kotlin activity. This list had a bunch of sample unix timestamps for a day's operating hour time slots. The last element was the first time slot of the next day. This hardcoded list allows us to test our onResponse handler, since right now any request to the operatingHours endpoint will return an empty list. Then, we should also hardcode the currentTime variable within this function and play around with its value ensuring the results are as expected.

- The cases we must test are as follows. The case that the currentTime is lower than the first time slots open time. Next case, is that the currentTime falls into one of the time slots in the day meaning that the facility is open. Another case is that the currentTime falls between two time slots so the facility is closed and will open when it is the second of the two time slots' open time. Finally, the last case is that the time is past the last time slot of the day's close time. After ensuring that all these edge cases are met, we should be good to go. 

- **We must also make sure to test that the JSON parsing of the API response works as expected! In order to do this, we should actually mock the API response, so that we are testing the JSON parsing of the API response body.** 

Screenshot_20200614-003344

Updated screenshot (2020-06-27)
Screenshot_Flux_20200627-185603

### Notes 

In building this feature, I had to make a few minor changes in other parts of the code.

1. Moved parseTime function from jsonParser class to FluxUtil, since it made more sense being a util function for the entire project, plus I ended up using the function in FacilityInfoPage.
2. Edited getCurrentDate() and getDayStringDaysAfter() functions in FluxUtil to ask for param specifying what format we want the date returned in, with year first or last (MM-DD-YYYY or YYYY-MM-DD). However this is something we should standardize across all our backend api endpoints. Thus, this is jus a temporary change!

### CODE REVIEW

File: Density/app/src/main/java/org/cornelldti/density/density/util/FluxUtil.kt [hunk 1]
```diff
@@ -57,19 +59,54 @@ object FluxUtil {
...
+        calendar.add(Calendar.DAY_OF_YEAR, daysAfter)
+        return calendar.time
+    }
+
     /**
      * getCurrentDate() provides the current date for the menu endpoint request.
+     * @param yearBeginning true if the format has year at beginning, false otherwise
      */
-    fun getCurrentDate(): String {
+    fun getCurrentDate(yearBeginning: Boolean): String {
```
[the remark this answers is not in the document]
Comment (kungpaogao) [reply Density/app/src/main/java/org/cornelldti/density/density/util/FluxUtil.kt:90]: also, there's another function called `getDate()` that we should probably combine

File: Density/app/src/main/java/org/cornelldti/density/density/util/FluxUtil.kt [hunk 2]
```diff
@@ -57,19 +59,54 @@ object FluxUtil {
...
+        else {
+            format = SimpleDateFormat("MM-dd-yyyy", Locale.US)
+        }
         return format.format(current.time)
     }
+
+    fun parseTime(timestamp: Long): String {
+        val timeZone = Calendar.getInstance().timeZone
+        var format = SimpleDateFormat("h:mma", Locale.US)
+        if (DateFormat.is24HourFormat(DensityApplication.getAppContext())) {
```
Comment (kungpaogao) [inline Density/app/src/main/java/org/cornelldti/density/density/util/FluxUtil.kt:105]: love me some 24h time 😍

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt [hunk 3]
```diff
@@ -200,53 +207,82 @@ class FacilityInfoPage : BaseActivity() {
...
+        api.facilityHours(facilityId=facilityClass!!.id, startDate = FluxUtil.convertDateObjectToString(date),
+                endDate = FluxUtil.convertDateObjectToString(nextDay),
+        facilityHoursTimeStampsOnResponse = {
+            hoursTimeStampsList ->
+            opHoursTimestamps = hoursTimeStampsList
+        },
+        facilityHoursStringsOnResponse = {
+            hoursStringsList ->
+            opHoursStrings = hoursStringsList
+        })
```
Comment (kungpaogao) [inline Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:224]: will things get messed up if `opHoursStrings` is not set in time for the UI updates?

Comment (kungpaogao) [reply Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt:224]: I guess it won't throw an error since it's just an empty list...

File: Density/app/src/main/java/org/cornelldti/density/density/util/FluxUtil.kt [hunk 4]
```diff
@@ -33,22 +33,33 @@ object FluxUtil {
     }
 
     /**
-     * getDate(day) provides the date for historical endpoint request.
+     * getDateString(day) provides the date in string format for historical endpoint request.
      */
-    fun getDate(day: String): String {
+    fun getDateObject(day: String): Date {
```
Comment (kungpaogao) [inline Density/app/src/main/java/org/cornelldti/density/density/util/FluxUtil.kt:38]: did the documentation get mixed up?

**Comment by kungpaogao** [comment]
Merging for now, but there is still a related issue #49.
